atef/check.py

```diff
@@ -780,3 +1071,28 @@ def get_signal_cache() -> _SignalCache[ophyd.EpicsSignalRO]:
     if _signal_cache is None:
         _signal_cache = _SignalCache(ophyd.EpicsSignalRO)
     return _signal_cache
+
+
+_yaml_initialized = False
+
+
+def _yaml_init():
+    """Add necessary information to PyYAML for serialization."""
+    global _yaml_initialized
+    if _yaml_initialized:
+        # Make it idempotent
+        return
+
+    _yaml_initialized = True
+
+    def int_enum_representer(dumper, data):
+        """Helper for pyyaml to represent enums as just integers."""
+        return dumper.represent_int(data.value)
+
+    def str_enum_representer(dumper, data):
+        """Helper for pyyaml to represent string enums as just strings."""
+        return dumper.represent_str(data.value)
+
+    # The ugliness of this makes me think we should use a different library
```

Comment:
A few possibilities, zero or more could be right:
* I may be missing a simpler/easier option for this
* The pyyaml library just isn't very nice (and maybe we should take a look at like `ruamel.yaml` which seems popular?)
* Maybe we drop yaml entirely? I do like the sample file format better than JSON though

Reply:
Without this weirdness, the yaml file is not reasonably human editable, right?

If ruamel.yaml gave a natively human readable/writable output I'd be in favor of switching.
I don't think it matters for this PR though.

I don't have an answer to dropping yaml or keeping it. It seems annoying to have multiple formats but it also doesn't seem like that much of a problem... I was thinking about making the GUI read both and save json by default or yaml if it sees .yml or .yaml

Reply:
Actually maybe the inverse is better: yaml by default, and the ability to switch to json in case your config file gets large enough to have a slow load.

Reply:
> Without this weirdness, the yaml file is not reasonably human editable, right?

It has weird 'tags' in there indicating how to recreate the Python enum object. "Human editable with frustration" may be appropriate.

A couple details I neglected to note:
* With those enum tags, `yaml.safe_load` doesn't work as it might allow for arbitrary code execution.
* As integer values in the yaml file, apischema dutifully converts these for us on deserialization, so storage as an integer is not a problem.

> If ruamel.yaml gave a natively human readable/writable output I'd be in favor of switching. I don't think it matters for this PR though.

I'd like to investigate it to see how much better it actually is. Agreed on being out of scope here.

> Actually maybe the inverse is better: yaml by default, and the ability to switch to json in case your config file gets large enough to have a slow load.

Yeah, that could be a reasonable approach.

I think we're both on the same page for "let's not make any big decisions on file formats just yet". Before we have tons of atef configuration files in the wild, though, we really should figure it out.